domain/src/main/java/com/aconno/sensorics/domain/ifttt/outcome/InputToActionsResolverImpl.kt

```diff
@@ -7,29 +7,30 @@ import io.reactivex.disposables.Disposable
 class InputToActionsResolverImpl(
     actionsRepository: ActionsRepository
 ) : InputToActionsResolver {
-    private var disposables : MutableList<Disposable> = mutableListOf()
+    private var actionsProviderDisposable : Disposable? = null
 
     private var actionsMap : MutableMap<ResolverKey,MutableList<Action>> = mutableMapOf()
 
     init {
         actionsRepository.getAllActionsAsFlowable().subscribe {
-            actionsMap = actionsToMap(it)
-        }.also { disposables.add(it) }
+            updateMapWithActions(it)
+        }.also { actionsProviderDisposable = it }
     }
 
-    private fun actionsToMap(actions : List<Action>) : MutableMap<ResolverKey,MutableList<Action>> {
-        return mutableMapOf<ResolverKey,MutableList<Action>>().apply {
+    private fun updateMapWithActions(actions : List<Action>) {
+        actionsMap.apply {
```

Comment:
Yeah, I forgot to remove this `apply` method. I have previously called it on new instance of map (`mutableMapOf().apply`) but when I changed `mutableMapOf` to `actionsMap`, I did not see that it would now not be appropriate to use `apply `method. I am going to fix this.